### DISCUSSION TITLE
WPF click transition

### DISCUSSION THREAD

**Discussion by CornyVanBeck** [AvaloniaUI/Avalonia#9465] [Q&A] [answered]
Hey everybody,

I would like to create a simple button animation on click on an ellipse. "pointerover" is working fine. Is there any possibility to do the same for a mouse click?

Minimal working example:
`<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:BigRedButton.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="300" d:DesignHeight="300"
		Height="300" Width="300"
        x:Class="BigRedButton.Views.MainWindow"
        Icon="/Assets/avalonia-logo.ico"
        Title="BigRedButton">

	<Design.DataContext>
		<vm:MainWindowViewModel/>
	</Design.DataContext>

	<Canvas Height="300" Width="300" Margin="150,150,0,0">
		<Ellipse Fill="Gray" Width="165" Height="155" Margin="-8,10,0,0"/>
		<Ellipse Fill="Silver" Width="155" Height="155" Margin="-3,5,0,0"/>
		<Ellipse Fill="DarkRed" Width="150" Height="150"/>
		<Ellipse Fill="Red" Width="150" Height="150">
			<Ellipse.Styles>
				<Style Selector="Ellipse">
					<Setter Property="RenderTransform" Value="translateY(-15px)"/>
				</Style>
				<Style Selector="Ellipse:pointerover">
					<Setter Property="RenderTransform" Value="translateY(0px)"/>
				</Style>
			</Ellipse.Styles>
			<Ellipse.Transitions>
				<Transitions>
					<TransformOperationsTransition Property="RenderTransform" Duration="0"/>
				</Transitions>
			</Ellipse.Transitions>
		</Ellipse>
	</Canvas>
</Window>
`

**Comment by Mrxx99** [comment]
You can try with [Avalonia.Xaml.Behaviors](https://github.com/wieslawsoltes/AvaloniaBehaviors). Look at the [ChangePropertyAction](https://github.com/wieslawsoltes/AvaloniaBehaviors/wiki/ChangePropertyAction) or the [AddClassAction](https://github.com/wieslawsoltes/AvaloniaBehaviors/blob/master/samples/BehaviorsTestApplication.Base/Views/Pages/AddRemoveClassActionView.axaml)

**Reply by CornyVanBeck** [reply]
Isn't that somehow possible with vanilla avalonia?

**Comment by rabbitism** [comment]
What about `:pressed`?

**Reply by CornyVanBeck** [reply]
`:pressed` does not work either, tryed that already

**Comment by MosaIssei** [comment, marked as answer]
use of button

	<Canvas Height="300" Width="300" Margin="150,150,0,0">
		<Ellipse Fill="Gray" Width="165" Height="155" Margin="-8,10,0,0"/>
		<Ellipse Fill="Silver" Width="155" Height="155" Margin="-3,5,0,0"/>
		<Ellipse Fill="DarkRed" Width="150" Height="150"/>
		<Button Width="150" Height="150" Focusable="False" Padding="0" Background="Transparent">
			<Button.Styles>
				<Style Selector="Button">
					<Setter Property="Background" Value="Transparent"/>
					<Setter Property="RenderTransform" Value="translateY(-15px)"/>
				</Style>
				<Style Selector="Button:pressed">
					<Setter Property="RenderTransform" Value="translateY(0px)"/>
				</Style>

				<Style Selector="Button:pointerover /template/ ContentPresenter">
					<Setter Property="Background" Value="Transparent"/>
				</Style>
			</Button.Styles>
			<Button.Transitions>
				<Transitions>
					<TransformOperationsTransition Property="RenderTransform" Duration="0:0:0.05"/>
				</Transitions>
			</Button.Transitions>
			<Ellipse Fill="Red">
			</Ellipse>
		</Button>
	</Canvas>

**Reply by CornyVanBeck** [reply]
Thank you very mutch, didn't know that the elements are combinable this way :)
